pkg/kubectl/cmd/attach.go

```diff
@@ -167,7 +167,7 @@ func (p *AttachOptions) Run() error {
 		if err != nil {
 			return err
 		}
-		if pod.Status.Phase != api.PodRunning {
+		if pod.Status.Phase != api.PodRunning && (pod.Status.Phase == api.PodPending && len(pod.Spec.InitContainers) < 1) {
```

Comment:
So in general we are not doing this for a lot of reasons (we are actively removing these tests).

Basically, @sttts has a PR that adds the right check - we get the pod and if it's in Succeeded or Failed we print an error, otherwise we leave the Kubelet to write the correct message